src/main/kotlin/de/craftan/engine/map/graph/Edge.kt

```diff
@@ -0,0 +1,104 @@
+package de.craftan.engine.map.graph
+
+import de.craftan.engine.map.GameTile
+import de.craftan.engine.map.TileDirection
+
+/**
+ * An edge connects the nodes and lies between up to 2 tiles
+ */
+data class Edge(
+    val tiles: Set<GameTile>,
+    val nodes: Pair<Node, Node>,
+    val structureInfo: StructureInfo,
+)
+
+enum class EdgeDirection(
+    val nodeDirections: Pair<NodeDirection, NodeDirection>,
+    val otherNode: TileDirection,
+) {
+    NORTH_WEST(Pair(NodeDirection.NORTH, NodeDirection.NORTH_WEST), TileDirection.NORTH_WEST),
+    NORTH_EAST(Pair(NodeDirection.NORTH, NodeDirection.NORTH_EAST), TileDirection.NORTH_EAST),
+    EAST(Pair(NodeDirection.NORTH_EAST, NodeDirection.SOUTH_EAST), TileDirection.EAST),
+    SOUTH_EAST(Pair(NodeDirection.SOUTH, NodeDirection.NORTH_EAST), TileDirection.SOUTH_EAST),
+    SOUTH_WEST(Pair(NodeDirection.SOUTH, NodeDirection.NORTH_WEST), TileDirection.SOUTH_WEST),
+    WEST(Pair(NodeDirection.NORTH_WEST, NodeDirection.SOUTH_WEST), TileDirection.WEST),
+}
+
+/**
+ * Generates the edges for given gametiles
+ * The gametiles MUST already have nodes
+ * @param tiles to generate edges for: MUST HAVE NODES
+ */
+fun generateEdges(tiles: List<GameTile>) {
+    for (tile in tiles) {
+        var previousNodeDirection = NodeDirection.NORTH_WEST
+        var previousNode = tile.nodes[previousNodeDirection]!!
+        for (nodeDirection in NodeDirection.entries) {
+            val node = tile.nodes[nodeDirection]!!
+
+            appendEdge(tiles, tile, previousNode, node, previousNodeDirection, nodeDirection)
```

Comment:
maybe move all of the helper functions, either to a file, or directly into the correct classes. So you could just do, tile.appendEdge() etc.